### DISCUSSION TITLE
New features: surfaces, moving windows out of the main window, multiple trajectories

### DISCUSSION THREAD

**Discussion by Eduardneu** [scanberg/viamd#61] [Ideas]
Hey guys, 

is just wanted to know whether these features are planned to be implemented or if you already have a certain release date in your mind: 

- representing selected residues as surfaces

- dragging the windows for "Animation", "Representations", "Timelines" out of the main window

- load multiple systems/multiple trajectories into a session

I think these features might be very useful for the community and would help to promote VIAMD even further.

Best
Eddie

**Comment by scanberg** [comment]
Hey Eddie,

We have discussed the features at some point in time, but I want to give you and any other people who may read this in the future an official response:

- representing selected residues as surfaces
Surface representations are sexy, I agree :). It has just fallen off the list of priorities as it was not deemed necessary from the beginning. And it would require some R&D as the state-of-the-art is perhaps a bit diffuse. We would like to transition away from OpenGL, so I avoid putting time and effort into that part for now.

- dragging the windows for "Animation", "Representations", and "Timelines" out of the main window
You can enable this from Cmake, if you use the GUI, it should be listed as VIAMD_IMGUI_ENABLE_VIEWPORTS to enable moving windows out from the main window. There is also VIAMD_IMGUI_ENABLE_DOCKSPACE to enable the docking of windows to the main viewport. These features are not considered stable for all platforms and have been causing some problems for me on Windows as well.

- load multiple systems/multiple trajectories into a session
I agree with your observation here and would personally rate this very high in major features to implement.
Some details need to be sorted out conceptually:
- Should the systems be put in the same 'workspace'/'session' or be separated by some 'tabs' where each tab represents a system?
- Should one topology be allowed to have multiple trajectories (ensembles) or would each topo + traj constitute a unique new system?
- What data should be allowed to be shared between the systems? e.g. computed properties and other artifacts.
- Should a script be individual to a system or globally evaluated for many systems (where each system is given an identifier to be referenced within the script)
- Etc.
I could go on for some time, but each point would further propagate into smaller design decisions that would have to be addressed.

So the short answer is: At some point in time :)

Best regards,
Robin
